### PR TITLE
Refactored debug_extra to be on ChoiceCaller (and called description)

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137410
* #137861
* #137859
* __->__ #137857
* #137768

Before:
image

After:
image

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -4024,11 +4022,12 @@ class ChoiceCaller:
     Children classes: TritonTemplateCaller, CUDATemplateCaller.
     """
 
-    def __init__(self, name, input_nodes, layout):
+    def __init__(self, name, input_nodes, layout, description):
         super().__init__()
         self.name = name
         self.layout = layout
         self.input_nodes = input_nodes
+        self.description = description
```
Comment (ezyang) [inline torch/_inductor/ir.py:4030]: Do you mind documenting it? A type would also be good

**Review by ezyang — APPROVED** [review]

File: torch/_inductor/ir.py (hunk 1 above)
Comment (Chillee) [reply torch/_inductor/ir.py:4030]: yeah it's just a string

**Review by jansel — APPROVED** [review]

**Review by masnesral — APPROVED** [review]

**Pull request closed without merging**